### PR TITLE
chore(code, deps): cleaned up some code, updated some dependencies

### PR DESCRIPTION
**Pull request by DuCanhGH**
**Brief Description of the Changes**
Give a concise description of what has changed.
This PR upgrades some dependencies, removes `@types/react` as it is no longer needed, and cleans up some code (replacing some deprecated APIs (of highlight.js, Typescript) with alternatives).

**Additional Notes for Testing**
Anything notable needed to do, starting from the standard test database?

### CODE REVIEW

File: frontend/ts/index.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ import hd from "humanize-duration";
 })();
 
 // Set timezone
-(function () {
+(() => {
```
Comment (natsukagami) [inline frontend/ts/index.ts:16]: Same here

File: frontend/ts/submission.ts [hunk 2]
```diff
@@ -22,19 +29,17 @@ hs.initHighlightingOnLoad();
               penalty: number;
           };
 
-    async function fetchResult(id: string): Promise<Result> {
-        return (
-            await fetch(`/admin/submissions/${id}/verdict`)
-        ).json() as Promise<Result>;
-    }
+    const fetchResult = async (id: string): Promise<Result> => {
```
Comment (natsukagami) [inline frontend/ts/submission.ts:32]: I'm not really sure why this is better?

File: frontend/package.json [hunk 3]
```diff
@@ -29,8 +28,9 @@
         "test": "echo No tests... yet",
         "fmt": "prettier \"css/**/*\" \"ts/**/*\" \"*.json\" \"*.js\"",
         "dev": "parcel watch \"html/**/*.html\"",
-        "build": "mkdir -p ../embed/templates && rm -rf ../embed/templates/* && parcel build --no-source-maps --no-cache \"html/**/*.html\"",
-        "build:windows": "pwsh --command ../scripts/windows/frontend_build.ps1"
+        "build": "mkdir -p ../embed/templates && rm -rf ../embed/templates/* && yarn build:parcel",
+        "build:parcel": "parcel build --no-source-maps --no-cache \"html/**/*.html\"",
+        "build:windows": "pwsh --command ../scripts/windows/frontend_cleanup.ps1 && yarn build:parcel"
```
Comment (natsukagami) [inline frontend/package.json:33]: Can pwsh handle &&?

Comment (DuCanhGH) [reply frontend/package.json:33]: @natsukagami defo, I've always used that.

File: frontend/ts/submission.ts (hunk 2 above)
Comment (DuCanhGH) [reply frontend/ts/submission.ts:32]: @natsukagami this avoids duplicating return types. Also it seems that a large part of the codebase uses arrow function expressions anyways, so unifying the style would be better.

File: frontend/purgecss.config.js [hunk 4]
```diff
@@ -1,5 +1,12 @@
+// @ts-check
+/** @type {import("purgecss").UserDefinedOptions} */
 module.exports = {
-    content: ["html/**/*.html", "html/*.html", "ts/**/*.ts", "ts/**/*.tsx"],
```
Comment (natsukagami) [inline frontend/purgecss.config.js:2]: Prettier did this?

File: frontend/ts/index.ts (hunk 1 above)
Comment (DuCanhGH) [reply frontend/ts/index.ts:16]: @natsukagami it seems that both work just fine, so unifying the code style would be nice.

File: frontend/ts/announcements.ts [hunk 5]
```diff
@@ -60,16 +58,16 @@ document.announcements = (() => {
     };
 
     // Fetch announcements count
-    const fetchAnnouncements = () => {
+    const fetchAnnouncements = async () => {
```
[the remark this answers is not in the document]
Comment (DuCanhGH) [reply frontend/ts/announcements.ts:61]: @natsukagami I mean both work, and the `await` syntax just flows better imo. That can be changed back, though.

[the remark this answers is not in the document]
Comment (natsukagami) [reply frontend/ts/announcements.ts:61]: Not too much in favor of this, and I don't like too many random changes in a PR either; please roll back

File: frontend/purgecss.config.js (hunk 4 above)
Comment (DuCanhGH) [reply frontend/purgecss.config.js:2]: @natsukagami probably.

File: frontend/ts/announcements.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (DuCanhGH) [reply frontend/ts/announcements.ts]: @natsukagami done in [58f8ba9](https://github.com/natsukagami/kjudge/pull/96/commits/58f8ba96b917054f9cc2ac4b395358b7a8fc90e2)

File: frontend/ts/announcements.ts [hunk 6]
```diff
@@ -70,6 +70,7 @@ document.announcements = (() => {
     };
 
     setInterval(fetchAnnouncements, 10 * 1000);
+
```
Comment (natsukagami) [inline frontend/ts/announcements.ts:73]: 😁 pls revert too

File: frontend/tailwind.config.js [hunk 7]
```diff
@@ -1,9 +1,7 @@
 import colors from "tailwindcss/colors";
 import defaultTheme from "tailwindcss/defaultTheme";
 
-/**
- * @type {import("tailwindcss").Config}
- */
+/** @type {import("tailwindcss").Config} */
```
Comment (natsukagami) [inline frontend/tailwind.config.js:4]: Do we need `@ts-check` here too like postcss.js

File: frontend/ts/submission.ts [hunk 8]
```diff
@@ -1,14 +1,21 @@
...
 
-hs.initHighlightingOnLoad();
+hs.registerLanguage("cpp", hsCpp);
+hs.registerLanguage("pascal", hsPas);
+hs.registerLanguage("go", hsGo);
+hs.registerLanguage("java", hsJava);
+hs.registerLanguage("python", hsPy);
+hs.registerLanguage("rust", hsRust);
+
+hs.highlightAll();
```
Comment (natsukagami) [inline frontend/ts/submission.ts:18]: Why not `initHighlightingOnLoad`?

Comment (natsukagami) [reply frontend/ts/submission.ts:18]: Hmm deprecated ok https://highlightjs.readthedocs.io/en/latest/api.html#inithighlightingonload

File: frontend/ts/announcements.ts (hunk 6 above)
Comment (DuCanhGH) [reply frontend/ts/announcements.ts]: @natsukagami done in [40d3a00](https://github.com/natsukagami/kjudge/pull/96/commits/40d3a003354ee632d49f90cbfba1583a98359f2a) :)

File: frontend/tailwind.config.js (hunk 7 above)
Comment (DuCanhGH) [reply frontend/tailwind.config.js:4]: @natsukagami it sorta helps improve UX so yeah (IDEs will report if we pass an invalid configuration).

**Review by natsukagami — APPROVED** [review]
Looks good now, thanks a lot <3

**Pull request merged**